src/theme/components/template.ts

```diff
@@ -1,111 +1,98 @@
-// From @mixin breakout:
-// Used for elements that should be edge-to-edge.
+/**
+ * Grid layout based on https://www.joshwcomeau.com/css/full-bleed/
+ */
+
 const breakout = {
-  marginEnd: "calc(-50vw + 50%)",
-  marginStart: "calc(-50vw + 50%)",
-};
-// From the `wrapper` SCSS mixin.
-// TODO: Eventually we may be able to put shared JS style objects
-// into a utils file for other mixins that are shared.
-const wrapperStyles = {
-  marginY: 0,
-  marginX: "auto",
-  maxWidth: "1280px",
-  paddingTop: 0,
-  paddingBottom: 0,
-  paddingEnd: "s",
-  paddingStart: "s",
   width: "100%",
+  // This could be "1 / 4" and it would mean the same. This is
+  // "future-proof" the grid column assignment to the last column.
+  gridColumn: "1 / -1",
 };
 
 const Template = {
   baseStyle: {
     boxSizing: "border-box",
     color: "ui.black",
-    // Users with non-overlay scrollbars have a small horizontal scrollbar from
-    // our breakout mixin. Fix identified here: https://cloudfour.com/thinks/breaking-out-with-viewport-units-and-calc/#one-big-dumb-caveat
-    overflowX: "hidden",
-    "*": { boxSizing: "inherit" },
-    "*::after": { boxSizing: "inherit" },
-    "*::before": { boxSizing: "inherit" },
+    display: "grid",
+    gridTemplateColumns: `
+      1fr
+      min(1280px, 100%)
+      1fr`,
+    // Set all elements to start on the second 1280px grid column.
+    "> *": {
+      gridColumn: "2",
+    },
   },
   sizes: {},
   defaultProps: {},
 };
-const TemplateHeader = {
-  baseStyle: {
-    marginBottom: "page.hstack", // --nypl-space-l
-  },
-};
+// Elements that need to breakout will span outside
+// the center 1280px grid column.
 const TemplateBreakout = {
-  baseStyle: {
-    ...breakout,
-    width: "100vw",
-    marginBottom: "page.hstack", // --nypl-space-l
-  },
+  baseStyle: breakout,
 };
 const TemplateContent = {
   baseStyle: {
-    ...wrapperStyles,
-    display: "block",
-    marginBottom: "page.hstack", // --nypl-space-l
+    display: "grid",
+    gridTemplateColumns: "1fr",
+    marginY: "page.hstack",
+    paddingY: 0,
+    paddingX: "s",
   },
+  // With left or right sidebars, we need to set two grid columns and
+  // the column for the sidebar is max 288px width.
   variants: {
-    sidebar: {
-      display: "flex",
-      flexFlow: {
-        base: "column nowrap",
-        md: "row wrap",
-      },
+    left: {
+      gridTemplateColumns: { md: "288px 1fr" },
```

Comment:
Should this be 287px?  255 + 32

Regardless, can we use `gap` and allow the width of the sidebar to be set to `255px`?  Including the gap value as part of hardcoded width value assumes that the grid gap will never change.

Reply:
When I set it to 255px, the width is smaller than what I expected. This was when I compared the flexbox sidebar with the grid sidebar in Turbine. I'll try setting the gap value but I'm not sure it'll render as expected.

Reply:
The gap works.